src/librustc_mir/interpret/eval_context.rs

```diff
@@ -548,7 +548,7 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                 Ok(Some(self.read_size_and_align_from_vtable(vtable)?))
             }
 
-            ty::Slice(_) | ty::Str => {
```

Comment:
potential site for the perf regression, easy to test by doing the old logic if `ty.is_str()`